packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/platform_interface/google_maps_flutter_platform.dart

```diff
@@ -360,6 +374,11 @@ abstract class GoogleMapsFlutterPlatform extends PlatformInterface {
     throw UnimplementedError('onLongPress() has not been implemented.');
   }
 
+  /// A marker icon managed by [ClusterManager] has been tapped.
+  Stream<ClusterTapEvent> onClusterTap({required int mapId}) {
```

Comment:
Ignore if this is your teams best practice. That said, throwing an error is a really aggressive way of dealing with apis that are not yet implemented as they can cause the app to crash.

Reply:
This is the standard way we define the platform interface base classes. We want people to know if they are trying to use functionality that hasn't been implemented for their platform yet rather than have it silently fail.

Reply:
Ok at some point we should talk about this being the best practice. As an app maintainer we did this on the flutter teams recommendation and found that it was extremely painful when adding platforms and changed our internal recommendation.